simulatingrisk/hawkdove/model.py

```diff
@@ -66,18 +66,50 @@ def initial_choice(self, hawk_odds=None):
     def choice_label(self):
         return "hawk" if self.choice == Play.HAWK else "dove"
 
+    def get_neighbors(self, size):
+        """get all neighbors for a supported neighborhood size"""
+        check_neighborhood_size(size)
+        # 4 and 8 neighborhood use default radius 1
+        # 8 and 24 both use moore neighborhood (includes diagonals)
+        opts = {"moore": True}
+        if size == 4:
+            # use von neumann neighborhood instead of moore (no diagonal)
+            opts["moore"] = False
+
+        # for 24 size neighborhood, use radius 2
+        if size == 24:
+            opts["radius"] = 2
+
+        return self.model.grid.get_neighbors(self.pos, include_center=False, **opts)
+
     @property
-    def neighbors(self):
-        # use configured neighborhood (with or without diagonals) on the model;
-        # don't include the current agent
-        return self.model.grid.get_neighbors(
-            self.pos, moore=self.model.include_diagonals, include_center=False
-        )
+    def play_neighbors(self):
+        """neighbors to play against, based on model play neighborhood size"""
+        return self.get_neighbors(self.model.play_neighborhood)
+
+    @property
+    def observed_neighbors(self):
+        """neighbors to look at when deciding what to play;
+        based on model observed neighborhood size"""
+        return self.get_neighbors(self.model.observed_neighborhood)
 
     @property
     def num_dove_neighbors(self):
-        """count how many neighbors played DOVE on the last round"""
-        return len([n for n in self.neighbors if n.last_choice == Play.DOVE])
+        """count how many neighbors played DOVE on the last round
+        (uses `observed_neighborhood` size from model)"""
+        return len([n for n in self.observed_neighbors if n.last_choice == Play.DOVE])
+
+    @property
+    def proportional_num_dove_neighbors(self):
+        """adjust the number of dove neighbors based on ratio between
+        play neighborhood and observed neighborhood, to scale observations
+        to the range of agent risk level."""
+        ratio = (
+            min(self.model.max_risk_level, self.model.play_neighborhood)
+            / self.model.observed_neighborhood
+        )
+        # always round to an integer
+        return round(ratio * self.num_dove_neighbors)
```

Comment:
@LaraBuchak here's the new logic for adjusting observed neighborhood so it matches the range of risk attitudes